### PR TITLE
Add completedAt fields

### PR DESCRIPTION
**Pull request by Seyi-Baks**
[Ticket](https://www.notion.so/chayn/Add-completedAt-fields-299b600051db4bd3bdbdc8797afde8c8)

Add completedAt:datetime field to both the SessionUser and CourseUser models, which will be null until the course/session is completed, and then filled with the datetime when their completed actions happen (i.e. when compeleted=true)

### CODE REVIEW

File: src/course-user/course-user.service.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ export class CourseUserService {
   async completeCourse({ userId, courseId }: CourseUserDto): Promise<CourseUserEntity> {
     const courseUser = await this.courseUserRepository.findOne({ where: { userId, courseId } });
     courseUser.completed = true;
+    courseUser.completedAt = new Date().toISOString();
```
Comment (annarhughes) [inline src/course-user/course-user.service.ts:34]: why `.toISOString` here? i'd have thought we want to keep it as a date everywhere

**Review by annarhughes — COMMENTED** [review]
lets stick to Date type everywhere. 

when i looked to see how `createdAt` and `updatedAt` work, i noticed that `updatedAt` isn't actually implemented - this is supposed to change every time the record is updated. If its quick let's get it done, otherwise please create a ticket for it. the updating the `updatedAt` should be done on a high level, not in every endpoint/function

File: src/course-user/course-user.service.ts (hunk 1 above)
Comment (Seyi-Baks) [reply src/course-user/course-user.service.ts:34]: It's similar to the date format used by Postgres/Typeorm 

2011-10-05T14:48:00.000Z

**Comment by Seyi-Baks** [comment]
The updatedAt field does work. It's easier to notice when you take a look at the partnerAccess table. It updates when an access code is assigned to a user. It's updated automatically by Typeorm using `@UpdateDateColumn`

**Pull request merged**